core/src/components/range/range.tsx

```diff
@@ -642,7 +649,7 @@ Developers can dismiss this warning by removing their usage of the "legacy" prop
   }
 
   private get hasLabel() {
-    return this.el.querySelector('[slot="label"]') !== null;
+    return this.el.querySelector('[slot="label"]') !== null || this.label !== undefined;
```

Comment:
Can we check for the `label` being defined first? It will logically be less expensive than querying the DOM.